tests/modules/token_embedders/pretrained_transformer_mismatched_embedder_test.py

```diff
@@ -143,3 +143,43 @@ def test_token_without_wordpieces(self):
         assert not torch.isnan(bert_vectors).any()
         assert all(bert_vectors[0, 1] == 0)
         assert all(bert_vectors[1, 1] == 0)
+
+    def test_exotic_tokens_no_nan_grads(self):
+        token_indexer = PretrainedTransformerMismatchedIndexer("bert-base-uncased")
+
+        sentence1 = ["A", "", "AllenNLP", "sentence", "."]
+        sentence2 = ["A", "\uf732\uf730\uf730\uf733", "AllenNLP", "sentence", "."]
+
+        tokens1 = [Token(word) for word in sentence1]
+        tokens2 = [Token(word) for word in sentence2]
+        vocab = Vocabulary()
+        params = Params(
+            {
+                "token_embedders": {
+                    "bert": {
+                        "type": "pretrained_transformer_mismatched",
+                        "model_name": "bert-base-uncased",
+                    }
+                }
+            }
+        )
+        token_embedder = BasicTextFieldEmbedder.from_params(vocab=vocab, params=params)
+
+        instance1 = Instance({"tokens": TextField(tokens1, {"bert": token_indexer})})
+        instance2 = Instance({"tokens": TextField(tokens2, {"bert": token_indexer})})
+
+        batch = Batch([instance1, instance2])
+        batch.index_instances(vocab)
+
+        padding_lengths = batch.get_padding_lengths()
+        tensor_dict = batch.as_tensor_dict(padding_lengths)
+        tokens = tensor_dict["tokens"]
+
+        bert_vectors = token_embedder(tokens)
+        test_loss = bert_vectors.mean()
+
+        test_loss.backward()
+
+        for name, param in token_embedder.named_parameters():
+            grad = param.grad
+            assert (grad is None) or (not torch.any(torch.isnan(grad)).item())
```

Comment:
In this case, something's wrong if we get `None` gradients, right?  So maybe just have the second case?

Reply:
I get `nan` gradients on `token_embedder_bert._matched_embedder.transformer_model.pooler.dense.weight` and `token_embedder_bert._matched_embedder.transformer_model.pooler.dense.bias`. Do you think this is a bug?

Reply:
Note: I get `None` for these even when I give well-behaved inputs with no missing or wonky tokens,

Reply:
Oh, ok, I didn't realize the pooler was included.  Yes, it's expected for the pooler to have `None` gradients, so we don't need to worry about that, and this is ok as it is.